### PR TITLE
improved "card_battery" card 

### PR DESCRIPTION
**Pull request by vncnt-dev**
I improved the "card_battery" card for users of the official Home Assistant Companion App, which indicate whether and how the device is being charged and colors the icon based on the battery level.
(no braking changes)

  wireless charging      |  wired charging
:-------------------------:|:-------------------------:
image| image

warning color     |  danger color
:-------------------------:|:-------------------------:
image| image

example config:
```yaml
- type: 'custom:button-card'
  template: card_battery
  entity: sensor.mi_9_battery_level
  variables:
       ulm_card_battery_battery_charger_type_entity_id: sensor.mi_9_charger_type
       ulm_card_battery_battery_battery_level_danger: 30
       ulm_card_battery_battery_battery_level_warning: 80
  name: Smartphone
```

<table>
<tr>
<th>Variable</th>
<th>Example</th>
<th>Required</th>
<th>Explanation</th>
</tr>
<tr>
<td>entity</td>
<td>sensor.yourSmartphone_battery_level</td>
<td>true</td>
<td>This is your battery entity</td>
</tr>
<tr>
<td>ulm_card_battery_attribute</td>
<td>battery_attribute</td>
<td>false</td>
<td>If your entity provides the battery percent in an attribute (= not as an own sensor), fill in the attribute's name here. E.g. if you have `sensor.livingroom_thermometer` and the attribute for your battery power is `sensor.livingroom_thermometer.attributes.battery_percent`, you fill in battery_percent here.</td>
</tr>
<tr>
<td>ulm_card_battery_battery_state_entity_id</td>
<td>sensor.yourSmartphone_battery_state</td>
<td>false</td>
<td>Entity that holds the battery state (charging/discharging). If provided, the Icon will display the current status.</td>
</tr>
<tr>
<td>ulm_card_battery_charger_type_entity_id</td>
<td>sensor.yourSmartphone_charger_type</td>
<td>false</td>
<td>Entity that holds the charger type (ac/wireless/none).
This Entity replaces the need for the `ulm_card_battery_battery_state_entity_id` entity.
 If provided, the Icon will display the current charger type. <br> This is only useful if you charge your devices Wireless and with cable.</td>
</tr>
<tr>
<td>ulm_card_battery_battery_level_danger</td>
<td>30</td>
<td>true</td>
<td>Changes the color of the Icon, if  the battery level falls below the provided value. Must be higher than `ulm_card_battery_battery_level_waring`</td>
</tr>
<tr>
<td>ulm_card_battery_battery_level_waring</td>
<td>50</td>
<td>true</td>
<td>Changes the color of the Icon, if the battery level falls below the provided value.</td>
</tr>
<tr>
<td>ulm_card_battery_color_battery_level_danger: </td>
<td>'var(--google-yellow)'</td>
<td>false</td>
<td>Color of icon if battery level is within the 'danger' zone.
default: 'var(--google-red)'</td>
</tr>
<tr>
<td>ulm_card_battery_color_battery_level_warning:</td>
<td>'var(--google-yellow)'</td>
<td>false</td>
<td>Color of icon if battery level is within the 'warning' zone.
default:  'var(--google-yellow)'</td>
</tr>
<tr>
<td>ulm_card_battery_color_battery_level_ok: </td>
<td>'var(--google-green)'</td>
<td>false</td>
<td>Color of icon if battery level is not within the 'danger' or 'warning' zone.
default: 'var(--google-green)'</td>
</tr>
</table>

### CODE REVIEW

**Comment by CM000n** [comment]
This is a great work! Thank you very much!
Would you mind switching to DEV as a base branch? There have been some developments in the meantime that currently prevent a merge.

File: config/minimalist-templates/button_card_templates.yaml [hunk 1]
```diff
@@ -842,53 +842,107 @@ card_battery:
...
+          [[[
+              var color = 'rgba(var(--color-theme),0.9)';
+              var battery_level = variables.ulm_card_battery_attribute !== null
+              ? states[entity.entity_id].attributes[variables.ulm_card_battery_attribute]
+              : states[entity.entity_id].state;
+
+              // battery_level<battery_level_danger and battery_level_warning not set
+              color = variables.ulm_card_battery_battery_level_danger !== null 
+                      && variables.ulm_card_battery_battery_level_danger>=battery_level
+                      && variables.ulm_card_battery_battery_level_warning == null?
```
Comment (bavo) [inline config/minimalist-templates/button_card_templates.yaml:931]: Why does it matter whether the warning level has been set here? 
In my opinion, if the battery level is lower than danger level we should always choose the danger level color.

**Comment by bavo** [comment]
I've posted a small remark above. 

But the card looks awesome! Great work :muscle:

**Comment by vncnt-dev** [comment]
> I've posted a small remark above.
> 
> But the card looks awesome! Great work 💪

Thanks for the code review.
You're right, I fixed it and also made the code cleaner in general.

**Review by bavo — APPROVED** [review]

File: config/minimalist-templates/button_card_templates.yaml [hunk 2]
```diff
@@ -850,53 +850,103 @@ card_battery:
     action: "more-info"
   variables:
     ulm_card_battery_attribute:
+    ulm_card_battery_battery_state_entity_id:
+    ulm_card_battery_charger_type_entity_id:
+    ulm_card_battery_battery_level_danger:
+    ulm_card_battery_battery_level_waring:
```
Comment (checkerschaf) [inline config/minimalist-templates/button_card_templates.yaml:856]: Here's a small typo. I guess it should be `ulm_card_battery_battery_level_warning`

Comment (vncnt-dev) [reply config/minimalist-templates/button_card_templates.yaml]: thanks

**Comment by CM000n** [comment]
Hey @vncnt-dev. Seems that you have some trailing whitespaces and wrong quotet string values in your code.
See the last pre-commit hook check: https://github.com/UI-Lovelace-Minimalist/UI/runs/4550906638?check_suite_focus=true#step:4:53

**Comment by vncnt-dev** [comment]
I have one warning left, but I can't figure out what the error is?
Could somebody help? :)

**Comment by CM000n** [comment]
I have merged it now. It was the way of commenting on line 291

This...
```yaml
    color: "blue" # blue, red, green, yellow, purple, pink
```

should be commented like this:
```yaml
    # blue, red, green, yellow, purple, pink
    color: "blue"
```

**Pull request merged**